src/jit/unwindarm.cpp

```diff
@@ -379,6 +379,15 @@ void Compiler::unwindPushMaskInt(regMaskTP maskInt)
 #if defined(_TARGET_UNIX_)
     if (generateCFIUnwindCodes())
     {
+        // If we are pushing LR, we assume the last thing the epilog will do is pop PC,
```

Comment:
For non-Unix / non-CFI codes, we generate unique unwind code sequences for both prolog and every epilog, and then, if they match (that is, the epilog is the inverse of the prolog), we can "re-use" the prolog codes for the epilog. And they match most of the time. However, it's not guaranteed that we always generate "pop {pc}" when we generate "push {lr}". I think we can also generate "pop {lr}; bx lr". I'm not sure if there are other epilog sequences that also don't match. So it seems like this wouldn't work for all cases.

Is there a reason why CFI is "different", such that we absolutely guarantee the epilog is an exact inverse of the prolog?

Reply:
I don't think we need to guarantee that the epilog is the reverse of the prolog. The comment is misleading, I'll fix it to just say that during unwind we need to get the PC of the caller by looking at the LR that is pushed.